core/frontend/src/tile/TileAdmin.ts

```diff
@@ -588,7 +588,16 @@ export class TileAdmin {
   public async generateTileContent(tile: IModelTile): Promise<Uint8Array> {
     this.initializeRpc();
     const props = this.getTileRequestProps(tile);
-    return IModelTileRpcInterface.getClient().generateTileContent(props.tokenProps, props.treeId, props.contentId, props.guid);
+    const retrieveMethod = await IModelTileRpcInterface.getClient().generateTileContent(props.tokenProps, props.treeId, props.contentId, props.guid);
+    if (retrieveMethod === TileContentSource.ExternalCache) {

```

Comment:
Ideally, if the tile request was canceled before generateTileContent completed, we would stop here instead of retrieving content we no longer need.

Reply:
how do we know if the request is canceled? Should I look at `tile.loadStatus`?

Reply:
`tile.request?.isCanceled`. See TileRequest.ts.